src/Centreon/Domain/Entity/EntityCreator.php

```diff
@@ -301,7 +301,7 @@ private function createSetterMethod(string $property): string
      * @param string $camelCaseName Name in camelCase format
      * @return string Returns the name converted in snake case format
      */
-    private function convertCamelCaseToSnakeCase(string $camelCaseName): string
+    public static function convertCamelCaseToSnakeCase(string $camelCaseName): string
     {
         $snakeCaseName = '';
```

Comment:
```suggestion
        $snakeCaseName = strtolower(preg_replace('/(?<!^)[A-Z]/', '_$0', $camelCaseName));
```

Reply:
all other lines except the return, can be removed... But the suggestion isn't available on these ;)